app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -31,6 +31,7 @@ def send_welcome_notification
         return if received_notification?(welcome_broadcast) || commented_on_welcome_thread? || user.created_at > 3.hours.ago
 
         Notification.send_welcome_notification(user.id, welcome_broadcast.id)
+        # Please do not remove the @notification_enqueued = true here - it is necessary for the notifications to run in the proper order
```

Comment:
```suggestion
        # Setting @notification_enqueued here prevents us from sending a user two welcome notifications in one day.
```

It's not so much that the flag here is running the notifications in a specific order, it's preventing it from sending two in one day.